api/src/test/java/com/epam/pipeline/acl/datastorage/DataStorageApiServiceFileTest.java

```diff
@@ -0,0 +1,1209 @@
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING_MAP;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.TEST_STRING_SET;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.anyLong;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class DataStorageApiServiceFileTest extends AbstractDataStorageAclTest {
+
+    private final DataStorageListing dataStorageListing = DatastorageCreatorUtils.getDataStorageListing();
+    private final AbstractDataStorageItem dataStorageFile = DatastorageCreatorUtils.getDataStorageFile();
+    private final DataStorageDownloadFileUrl downloadFileUrl =
+            DatastorageCreatorUtils.getDataStorageDownloadFileUrl();
+    private final DataStorageItemContent dataStorageItemContent =
+            DatastorageCreatorUtils.getDefaultDataStorageItemContent();
+    private final TemporaryCredentials temporaryCredentials = DatastorageCreatorUtils.getTemporaryCredentials();
+
+    private final List<PathDescription> pathDescriptionList = DatastorageCreatorUtils.getPathDescriptionList();
+    private final List<String> testList = Collections.singletonList(TEST_STRING);
+    private final List<UpdateDataStorageItemVO> dataStorageItemVOList =
+            DatastorageCreatorUtils.getUpdateDataStorageItemVOList();
+    private final List<DataStorageFile> dataStorageFileList = DatastorageCreatorUtils.getDataStorageFileList();
+    private final List<DataStorageDownloadFileUrl> downloadFileUrlList =
+            DatastorageCreatorUtils.getDataStorageDownloadFileUrlList();
+    private final List<DataStorageAction> dataStorageActionList = DatastorageCreatorUtils.getDataStorageActionList();
+
+    @Autowired
+    private TemporaryCredentialsManager mockTemporaryCredentialsManager;
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldGetDataStorageItemsForAdmin() {
+        doReturn(dataStorageListing).when(mockDataStorageManager)
+                .getDataStorageItems(ID, TEST_STRING, true, TEST_INT, TEST_STRING);
+        mockUserContext(context);
+
+        assertThat(dataStorageApiService.getDataStorageItems(
+                ID, TEST_STRING, true, TEST_INT, TEST_STRING)).isEqualTo(dataStorageListing);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldGetDataStorageItemsWhenPermissionIsGranted() {
+        initAclEntity(s3bucket, AclPermission.READ);
+        doReturn(dataStorageListing).when(mockDataStorageManager)
+                .getDataStorageItems(ID, TEST_STRING, true, TEST_INT, TEST_STRING);
+        initUserAndEntityMocks(OWNER_USER, s3bucket, context);
```

Comment:
It looks a bit confusing: test runs with `SIMPLE_USER` environment but then mocks `OWNER_USER`.

Reply:
It still looks a bit confusing